x/stake/keeper/validator_test.go

```diff
@@ -1043,3 +1045,71 @@ func TestGetValidTendermintUpdatesBondTransition(t *testing.T) {
 	clearTendermintUpdates(ctx, keeper)
 	require.Equal(t, 0, len(keeper.GetValidTendermintUpdates(ctx)))
 }
+
+func TestUpdateValidatorCommission(t *testing.T) {
+	ctx, _, keeper := CreateTestInput(t, false, 1000)
+	ctx = ctx.WithBlockHeader(abci.Header{Time: time.Now().UTC()})
+
+	commission := types.NewCommission(sdk.NewDecWithPrec(1, 1), sdk.NewDecWithPrec(3, 1), sdk.NewDecWithPrec(1, 1))
+
+	val1 := types.NewValidator(addrVals[0], PKs[0], types.Description{})
+	val2 := types.NewValidator(addrVals[1], PKs[1], types.Description{})
+
+	val1, _ = val1.SetInitialCommission(commission, time.Now().UTC().Add(time.Duration(-1)*time.Hour))
+	val2, _ = val2.SetInitialCommission(commission, time.Unix(0, 0).UTC())
+
+	testCases := []struct {
+		validator   types.Validator
+		newRate     sdk.Dec
+		expectedErr bool
+	}{
+		{
+			validator:   val1,
+			newRate:     sdk.ZeroDec(),
+			expectedErr: true,
+		},
+		{
+			validator:   val2,
+			newRate:     sdk.NewDecWithPrec(-1, 1),
+			expectedErr: true,
+		},
+		{
+			validator:   val2,
+			newRate:     sdk.NewDecWithPrec(4, 1),
+			expectedErr: true,
+		},
+		{
+			validator:   val2,
+			newRate:     sdk.NewDecWithPrec(3, 1),
+			expectedErr: true,
+		},
+		{
+			validator:   val2,
+			newRate:     sdk.NewDecWithPrec(2, 1),
+			expectedErr: false,
+		},
```

Comment:
Although I typically am all for expanded object definitions to keep things extra explicit. In table driven tests like this, where the definition of the test struct is right here in the function, I think it's fine to have compressed definitions, to keep things more concise, also the definition is right there so we don't lose the context of "what is going where" . aka let's have one line definitions here  like `{val2,  sdk.NewDecWithPrec(2, 1), false}`